modules/Hlsjs/resources/hlsjs.js

```diff
@@ -473,10 +473,10 @@
 			/**
 			 * Override player callback after changing media
 			 */
-			changeMediaCallback: function(){
-				this.getPlayer().play();
```

Comment:
This change retuned the changeMediaCallback flow, and just overrides the actual vid.src switch with the hlsjs logic of switching.
All other logic should remains the same.
In any case, I verified it and it's OK. Thanks!